scripts/elasticsearch_nodes.py

```diff
@@ -0,0 +1,83 @@
+"""
+This script is referenced by TerraForm as an external data source.
+Prints the current deployment's Elasticsearch node IDs as JSON to STDOUT.
+"""
+from itertools import (
+    repeat,
+)
+import json
+import logging
+import sys
+from time import (
+    sleep,
+)
+
+from botocore.exceptions import (
+    ClientError,
+)
+
+from azul import (
+    config,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+# How many times to retry if the request to AWS returns an error.
+MAX_ATTEMPTS = 10
+
+# The number of nodes in the cloudwatch dashboard template file.
+NUM_TEMPLATE_NODES = 4
+
+
+def main():
+    for attempt in range(MAX_ATTEMPTS):
```

Comment:
This is currently misleading because the iterator is actually never exhausted, due to the look-ahead in the `if` statement below. It can be rewritten so that the loop variable is tested in one place only, and  the number of attempts is also mentioned only once. `i = iter(range(n)); while True; ... if next(i, None) is None ... else ...`